source/grid/grid_out.cc

```diff
@@ -2570,16 +2570,15 @@ void GridOut::write_debug_vtu_per_processor (const Triangulation<dim,spacedim> &
             patch.data(1,vertex) = -1.0;
           patch.data(2,vertex) =
             (double)static_cast<int>(cell->level_subdomain_id());
-          patch.data(3,vertex) =
-            (double)static_cast<int>(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD));
+          patch.data(3,vertex) = tria.locally_owned_subdomain();
         }
```

Comment:
Yes, but I think you will get a warning now because you removed the cast to double.